### PR TITLE
Adds analytics to track the thumbnail/preview mode selection

### PR DESCRIPTION
**Pull request by antonis**
**Fixes** https://github.com/wordpress-mobile/gutenberg-mobile/issues/3025

## Description
This PR adds analytics to track the thumbnail/preview mode (mobile, tablet, desktop) selection

## To test:

<details>
<summary><strong>To start the Home Page Picker screen</strong></summary>

1. Start the site creation flow (e.g. *Choose site > Add button*)
1. Select the *Create WordPress.com site* option
</details>

#### Note
An easy way to test for these analytics is to debug the app and filter Logcat to look for the string `🔵 Tracked:`

### Start Home Page Picker
1. **Start** the Home Page Picker
1. **Verify** that the `🔵 Tracked: enhanced_site_creation_site_design_viewed, Properties: {"thumbnail_mode":"mobile"}` event is emitted (the value would be `tablet` on a tablet)

### Thumbnail mode button press
1. Start the Home Page Picker
1. Press the **thumbnail mode button** next to the skip button
1. **Verify** that the `🔵 Tracked: enhanced_site_creation_site_design_thumbnail_mode_button_tapped, Properties: {"preview_mode":"mobile"}` event is emitted (where `mobile` is the current mode when the button is pressed)

### Change thumbnail mode
1. Start the Home Page Picker
1. Press the **thumbnail mode button** next to the skip button
1. Select a thumbnail mode
1. **Verify** that the `🔵 Tracked: enhanced_site_creation_site_design_preview_mode_changed, Properties: {"preview_mode":"tablet"}` event is emitted where `tablet` is the newly selected mode

### Preview
1. Start the Home Page Picker
1. Select a design
1. Press **Preview**
1. **Verify** that the `🔵 Tracked: enhanced_site_creation_site_design_preview_viewed, Properties: {"template":"alves","preview_mode":"desktop"}` event is emitted where `desktop` the current mode

### Preview mode button press
1. Start the Home Page Picker
1. Select a design
1. Press **Preview**
1. Press the **preview mode button** at the top right of the screen
1. **Verify** that the `🔵 Tracked: enhanced_site_creation_site_design_preview_mode_button_tapped, Properties: {"preview_mode":"desktop"}` event is emitted (where `desktop` is the current mode when the button is pressed)

### Change preview mode
1. Start the Home Page Picker
1. Select a design
1. Press **Preview**
1. Press the **preview mode button** at the top right of the screen
1. Select a thumbnail mode
1. **Verify** that the `🔵 Tracked: enhanced_site_creation_site_design_preview_mode_changed, Properties: {"preview_mode":"tablet"}` event is emitted where `tablet` is the newly selected mode

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/misc/SiteCreationTracker.kt [hunk 1]
```diff
@@ -13,6 +20,16 @@ enum class SiteCreationErrorType {
...
 @Singleton
 class SiteCreationTracker @Inject constructor(val tracker: AnalyticsTrackerWrapper) {
+    private enum class PROPERTY(val key: String) {
+        TEMPLATE("template"),
+        SEGMENT_NAME("segment_name"),
+        SEGMENT_ID("segment_id"),
+        CHOSEN_DOMAIN("chosen_domain"),
+        SEARCH_TERM("search_term"),
+        THUMBNAIL_MODE("thumbnail_mode"),
+        PREVIEW_MODE("preview_mode")
```
Comment (mkevins) [inline WordPress/src/main/java/org/wordpress/android/ui/sitecreation/misc/SiteCreationTracker.kt:30]: Good idea to extract these :smiley:

**Review by mkevins — COMMENTED** [review]
Code looks correct to me. I tested all the flows, and everything is mostly working as described. I encountered one issue for this test:

> ### Preview
> 1. Start the Home Page Picker
> 1. Select a design
> 1. Press **Preview**
> 1. **Verify** that the `🔵 Tracked: enhanced_site_creation_site_design_preview_viewed, Properties: {"template":"alves","preview_mode":"desktop"}` event is emitted where `desktop` the current mode

Which is that the `enhanced_site_creation_site_design_preview_loaded` event was tracked / emitted twice on some occasions. I noticed that there was an error in the logs nearby, which I beleive is related with the injected JS code (`"Uncaught TypeError: Cannot read property 'setAttribute' of null", source:  (1)`), so maybe it's somehow related to that. :man_shrugging: 

```
2021-02-01 14:40:06.106 15283-15283/org.wordpress.android.prealpha I/WordPress-STATS: 🔵 Tracked: enhanced_site_creation_site_design_thumbnail_mode_button_tapped, Properties: {"preview_mode":"tablet"}
2021-02-01 14:40:07.088 15283-15283/org.wordpress.android.prealpha I/WordPress-STATS: 🔵 Tracked: enhanced_site_creation_site_design_preview_mode_changed, Properties: {"preview_mode":"desktop"}
2021-02-01 14:40:16.464 15283-15283/org.wordpress.android.prealpha I/WordPress-STATS: 🔵 Tracked: enhanced_site_creation_site_design_preview_viewed, Properties: {"template":"alves","preview_mode":"desktop"}
2021-02-01 14:40:16.606 15283-15283/org.wordpress.android.prealpha I/WordPress-STATS: 🔵 Tracked: enhanced_site_creation_site_design_preview_loading, Properties: {"template":"alves","preview_mode":"desktop"}
2021-02-01 14:40:16.663 15283-15429/org.wordpress.android.prealpha W/ndroid.prealph: Accessing hidden method Landroid/media/AudioManager;->getOutputLatency(I)I (greylist, reflection, allowed)
2021-02-01 14:40:16.670 15283-15429/org.wordpress.android.prealpha W/cr_media: Requires BLUETOOTH permission
2021-02-01 14:40:16.717 15283-15283/org.wordpress.android.prealpha I/WordPress-STATS: 🔵 Tracked: enhanced_site_creation_site_design_preview_loading, Properties: {"template":"alves","preview_mode":"desktop"}
2021-02-01 14:40:17.013 15283-15283/org.wordpress.android.prealpha I/chromium: [INFO:CONSOLE(1)] "Uncaught TypeError: Cannot read property 'setAttribute' of null", source:  (1)
2021-02-01 14:40:17.018 15283-15283/org.wordpress.android.prealpha I/WordPress-STATS: 🔵 Tracked: enhanced_site_creation_site_design_preview_loaded, Properties: {"template":"alves","preview_mode":"desktop"}
2021-02-01 14:40:22.054 15283-15283/org.wordpress.android.prealpha I/WordPress-STATS: 🔵 Tracked: enhanced_site_creation_site_design_preview_loaded, Properties: {"template":"alves","preview_mode":"desktop"}
```

**Comment by antonis** [comment]
Hey @mkevins 👋 

Thank you for reviewing this 🙇 

> Which is that the `enhanced_site_creation_site_design_preview_loaded` event was tracked / emitted twice on some occasions.

I fixed this with https://github.com/wordpress-mobile/WordPress-Android/pull/13831/commits/a091a594eec58a6a34e21089ce0c88ce62bc98ce. It seems that it was caused because the preview loading was triggered twice the first time the preview screen appeared:
* one with the [`load()` call at the end of the `onViewCreated`](https://github.com/wordpress-mobile/WordPress-Android/blob/a091a594eec58a6a34e21089ce0c88ce62bc98ce/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/DesignPreviewFragment.kt#L144)
* and one [when the previewMode was initialized](https://github.com/wordpress-mobile/WordPress-Android/blob/69b33d08f568a07cc140fa05203d4b0be1911687/WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/HomePagePickerViewModel.kt#L100)

> I noticed that there was an error in the logs nearby, which I beleive is related with the injected JS code (`"Uncaught TypeError: Cannot read property 'setAttribute' of null", source: (1)`), so maybe it's somehow related to that. 🤷‍♂️

This was indeed related with the preview loading twice and does not occur after the fix.

**Comment by mkevins** [comment]
The latest changes resolve the issue with the double emission of tracked events. I'm not sure when this occurred, but I noticed that there are now only two options in the post preview from within the edit (as well as from the post card "view" option):



Is this expected?

**Comment by antonis** [comment]
Hello @mkevins 👋 

> The latest changes resolve the issue with the double emission of tracked events. 

thank you for retesting this 🙇 

> I'm not sure when this occurred, but I noticed that there are now only two options in the post preview from within the edit (as well as from the post card "view" option):

Since I did some minor refactoring in the preview mechanism as part of this PR I based the enhancement of the post/page and site previews with tablet mode on top of it. Thought it is [approved](https://github.com/wordpress-mobile/WordPress-Android/pull/13787) I haven't merge this functionality yet.

**Review by mkevins — APPROVED** [review]
This is looking good! I tested the flows on a Pixel 3a (physical device) and checked logcat, and everything is working as described. LGTM! :+1: 

> I haven't merge this functionality yet.

I couldn't see what about this PR would have reverted those changes..  :laughing: This makes perfect sense now.

**Pull request merged**